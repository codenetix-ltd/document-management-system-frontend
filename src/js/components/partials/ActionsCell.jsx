import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { DataLink } from 'Components/common/dataControls';

/**
 * Component for react-table's actions column with edit and delete buttons
 * @param props
 * @returns {*}
 * @constructor
 */
export default function ActionsCell(props) {
  const { editLink, rowData, onDelete } = props;
  return (
    <div>
      <Link to={editLink} className="btn-success btn btn-xs">
        <i className="fa fa-edit" />
      </Link>
      &nbsp;
      <DataLink cls="btn-danger btn btn-xs" data={rowData} onClick={onDelete}>
        <i className="fa fa-trash" />
      </DataLink>
    </div>
  );
}

ActionsCell.propTypes = {
  editLink: PropTypes.string.isRequired,
  rowData: PropTypes.any.isRequired,
  onDelete: PropTypes.func.isRequired
};
