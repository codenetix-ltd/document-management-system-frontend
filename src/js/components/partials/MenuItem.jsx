import React from 'react';
import cln from 'classnames';
import PropTypes from 'prop-types';
import { If, Then } from 'qc-react-conditionals/lib';
import { Link } from 'react-router-dom';

import { DataLink } from 'Components/common/dataControls';

export default function MenuItem(props) {
  const {
    name,
    active,
    iconCls,
    addLink,
    listLink,
    addLinkActive,
    listLinkActive
  } = props;
  return (
    <li className={cln({ treeview: true, active })}>
      <DataLink><i className={iconCls} /> <span>{name}</span></DataLink>
      <ul className="treeview-menu">
        <li className={cln({ active: listLinkActive })}>
          <Link to={listLink}>
            <i className="fa fa-list" /> <span>List all</span>
          </Link>
        </li>
        <If is={name !== 'Logs'}>
          <Then>
            <li className={cln({ active: addLinkActive })}>
              <Link to={addLink}>
                <i className="fa fa-plus" /> <span>Add new</span>
              </Link>
            </li>
          </Then>
        </If>
      </ul>
    </li>
  );
}

MenuItem.propTypes = {
  name: PropTypes.string.isRequired,
  iconCls: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  addLinkActive: PropTypes.bool.isRequired,
  listLinkActive: PropTypes.bool.isRequired,
  listLink: PropTypes.string.isRequired,
  addLink: PropTypes.string.isRequired
};
