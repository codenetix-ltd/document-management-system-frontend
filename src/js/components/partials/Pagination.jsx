import React, { Component } from 'react';
import cln from 'classnames';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';

import { DataLink } from 'Components/common/dataControls';

@autobind
export default class ReactTablePagination extends Component {
  static defaultProps = {
    page: 1,
    pages: -1,
    onPageChange: () => {},
    canPrevious: false,
    canNext: false,
    className: '',
    paginationStyle: null,
    previousText: '',
    nextText: ''
  };

  static propTypes = {
    page: PropTypes.number,
    pages: PropTypes.number,
    onPageChange: PropTypes.func,
    canPrevious: PropTypes.bool,
    canNext: PropTypes.bool,
    className: PropTypes.string,
    paginationStyle: PropTypes.any,
    previousText: PropTypes.string,
    nextText: PropTypes.string
  };

  onPrevClick() {
    const { canPrevious, page } = this.props;
    if (!canPrevious) return;
    this.changePage(page - 1);
  }

  onNextClick() {
    const { canNext, page } = this.props;
    if (!canNext) return;
    this.changePage(page + 1);
  }

  onPageClick(page) {
    this.changePage(page);
  }

  getSafePage(page) {
    if (Number.isNaN(page)) {
      page = this.props.page; // eslint-disable-line
    }
    return Math.min(Math.max(page, 0), this.props.pages - 1);
  }

  changePage(pageNum) {
    const page = this.getSafePage(pageNum);
    if (this.props.page !== page) {
      this.props.onPageChange(page);
    }
  }

  render() {
    const {
      pages,
      page,
      canNext,
      className,
      paginationStyle,
      previousText,
      nextText
    } = this.props;

    // todo: use https://github.com/ultimate-pagination/react-ultimate-pagination
    const pagesList = Array.from({ length: pages }, () => '').map((p, index) => {
      const thresholdMax = page + 5;
      const thresholdMin = thresholdMax - 5;
      const pageNum = index + 1;
      const cls = cln('paginate_button', { active: index === page });
      if (pageNum === pages) {
        return (
          <li key={index} className={cls}>
            <DataLink data={index} onClick={this.onPageClick}>{pageNum}</DataLink>
          </li>
        );
      }
      if (pageNum === thresholdMax) {
        return (
          <li key={index} className="paginate_button disabled">
            <DataLink>...</DataLink>
          </li>
        );
      }
      if (pageNum > thresholdMax || pageNum <= thresholdMin) return null;
      return (
        <li key={index} className={cls}>
          <DataLink data={index} onClick={this.onPageClick}>{pageNum}</DataLink>
        </li>
      );
    });

    return (
      <ul className={cln(className, ' pagination')} style={paginationStyle}>
        <li className={cln('paginate_button', 'previous', { disabled: !canNext })}>
          <DataLink onClick={this.onPrevClick}>{previousText}</DataLink>
        </li>
        {pagesList}
        <li className={cln('paginate_button', 'next', { disabled: !canNext })}>
          <DataLink onClick={this.onNextClick}>{nextText}</DataLink>
        </li>
      </ul>
    );
  }
}
