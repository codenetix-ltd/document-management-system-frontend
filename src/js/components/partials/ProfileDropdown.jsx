import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import autobind from 'autobind-decorator';
import cln from 'classnames';
import onClickOutside from 'react-onclickoutside';

import { DataLink } from 'Components/common/dataControls';

import { $$profileFetch } from 'Store/thunks/profile';
import { $$logOut } from 'Store/thunks/auth';

@autobind
export class ProfileDropdown extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    profile: PropTypes.any.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      userMenuOpen: false
    };
  }

  componentDidMount() {
    $$profileFetch(this.props.dispatch);
  }

  onLogout() {
    $$logOut(this.props.dispatch);
  }

  handleClickOutside() {
    this.setState({
      userMenuOpen: false
    });
  }

  toggleProfileDropdown() {
    this.setState({
      userMenuOpen: !this.state.userMenuOpen
    });
  }

  render() {
    const userMenuCls = cln('dropdown', 'user', 'user-menu', {
      open: this.state.userMenuOpen
    });
    const { profile } = this.props;
    return (
      <div className="navbar-custom-menu">
        <ul className="nav navbar-nav">
          <li className={userMenuCls}>
            <DataLink cls="dropdown-toggle" onClick={this.toggleProfileDropdown}>
              <img
                src={require('Images/user-default.jpeg')}
                className="user-image"
                alt=""
                width="25"
                height="25"
              />
              <span className="hidden-xs">{profile.fullName}</span>
            </DataLink>
            <ul className="dropdown-menu">
              <li className="user-header">
                <img
                  src={require('Images/user-default.jpeg')}
                  className="img-circle"
                  alt="User Image"
                />
                <p>{profile.fullName}</p>
              </li>
              <li className="user-footer">
                <div className="pull-left">
                  <Link
                    to="/profile"
                    className="btn btn-default btn-flat"
                    onClick={this.handleClickOutside}
                  >
                    Edit profile
                  </Link>
                </div>
                <div className="pull-right">
                  <button
                    type="button"
                    onClick={this.onLogout}
                    className="btn btn-default btn-flat"
                  >Logout
                  </button>
                </div>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    );
  }
}

const Wrapped = onClickOutside(ProfileDropdown);

const mapStateToProps = ({ profile }) => ({ profile });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(Wrapped);
