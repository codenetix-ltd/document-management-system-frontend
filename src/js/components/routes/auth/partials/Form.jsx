import React, { Component } from 'react';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';

import ls from 'Services/SecureLS';

@autobind
export default class AuthForm extends Component {
  static defaultProps = {
    submitButtonText: 'Login',
    validate: () => false
  };

  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    submitButtonText: PropTypes.string,
    validate: PropTypes.func
  };

  constructor(props) {
    super(props);
    const { email, password } = ls.get('c_cache');
    this.state = {
      email: email || '',
      password: password || '',
      remember: false
    };
  }

  /**
   * Handles submit event.
   * @param e
   */
  onSubmit(e) {
    e.preventDefault();
    const { email, password, remember } = this.state;
    if (remember) ls.set('c_cache', { email, password });
    this.props.onSubmit({ email, password });
  }

  /**
   * Universal handler for multiple input fields
   * @param event
   */
  handleChange(event) {
    const { target } = event;
    const { value, name } = target;
    this.setState({
      [name]: value
    });
  }

  toggleCheckbox() {
    this.setState({ remember: !this.state.remember });
  }

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <div className="form-group has-feedback">
          <input
            type="email"
            className="form-control"
            placeholder="Email"
            name="email"
            value={this.state.email}
            onChange={this.handleChange}
          />
          <span className="form-control-feedback">
            <i className="fa fa-envelope" />
          </span>
        </div>
        <div className="form-group has-feedback">
          <input
            type="password"
            className="form-control"
            placeholder="Password"
            name="password"
            value={this.state.password}
            onChange={this.handleChange}
          />
          <span className="form-control-feedback">
            <i className="fa fa-lock" />
          </span>
        </div>
        <div className="row">
          <div className="col-xs-8">
            <div className="checkbox">
              <label>
                <div className="icheckbox_square-blue">
                  <input
                    type="checkbox"
                    checked={this.state.remember}
                    onChange={this.toggleCheckbox}
                  />
                </div>
                Remember Me
              </label>
            </div>
          </div>
          <div className="col-xs-4">
            <button
              type="submit"
              className="btn btn-primary btn-block btn-flat"
              disabled={this.props.validate(this.state)}
            >
              {this.props.submitButtonText}
            </button>
          </div>
        </div>
      </form>
    );
  }
}
