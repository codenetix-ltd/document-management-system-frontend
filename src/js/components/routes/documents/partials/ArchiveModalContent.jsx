import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'react-select';
import { connect } from 'react-redux';
import autobind from 'autobind-decorator';
import pick from 'lodash/pick';

import { $substituteDocument } from 'Store/actions';

@autobind
export class ArchiveModalContent extends Component {
  static defaultProps = {
    substituteDocument: null
  };

  static propTypes = {
    substituteDocument: PropTypes.any,
    dispatch: PropTypes.func.isRequired,
    documents: PropTypes.any.isRequired,
    selectedDocuments: PropTypes.array.isRequired
  };

  onChange(value) {
    this.props.dispatch($substituteDocument(value));
  }

  render() {
    const {
      documents: { list },
      substituteDocument,
      selectedDocuments
    } = this.props;
    const options = list.map(({ actualVersion }) => actualVersion);
    const titlePart = selectedDocuments.length > 1 ? 'selected documents' : 'the document';
    return (
      <div>
        <p>{`Do you really want to archive ${titlePart}?`}</p>
        <hr />
        <form className="form-horizontal">
          <div className="form-group">
            <label className="col-md-4 control-label">New actual document</label>
            <div className="col-md-8">
              <Select
                options={options}
                labelKey="name"
                valueKey="id"
                value={substituteDocument}
                onChange={this.onChange}
              />
            </div>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (state) => pick(state, ['documents', 'substituteDocument', 'selectedDocuments']);

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(ArchiveModalContent);
