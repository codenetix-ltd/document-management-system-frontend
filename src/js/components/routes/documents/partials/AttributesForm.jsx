import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';

import {
  DeviationInput,
  DataInput as Input,
  DataCheckbox as Checkbox
} from 'Components/common/dataControls';

import { $updateAttributeValues } from 'Store/actions';

@autobind
export class AttributesForm extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    document: PropTypes.any.isRequired,
    types: PropTypes.array.isRequired
  };

  onAttrChange({ target }, data) {
    const { id } = data;
    const { value } = target;
    const { dispatch } = this.props;
    dispatch($updateAttributeValues([{ id, value }]));
  }

  onCheckboxToggle(e, data) {
    const { id } = data;
    const { dispatch, document: { actualVersion: { attributeValues } } } = this.props;
    const found = attributeValues.find(at => at.id === id);
    const value = found ? found.value : false;
    dispatch($updateAttributeValues([{ id, value: !value }]));
  }

  getType(typeId) {
    return this.props.types.find(type => type.id === typeId);
  }

  render() {
    const { document: { actualVersion } } = this.props;
    const { attributeValues: attrValues, template } = actualVersion;
    return (
      <table className="table table-bordered table-view-document">
        <tbody>
          {
            template.attributes.map((attr, index) => {
              const type = this.getType(attr.typeId);
              if (type.machineName === 'table') {
                if (!attr.data.length) {
                  console.trace('no data in attribute');
                  return null;
                }
                const { headers: cols, rows } = attr.data;
                return (
                  <tr key={index}>
                    <td>{attr.name}</td>
                    <td>
                      <table className="table table-bordered table-as-value">
                        <tbody>
                          <tr>
                            <td>&nbsp;</td>
                            {cols.map(({ name }, i) => <td key={i}><b>{name}</b></td>)}
                          </tr>
                          {
                            rows.map(({ name, columns }, idx) => {
                              return (
                                <tr key={idx}>
                                  <td>{name}</td>
                                  {
                                    columns.map((col, i) => {
                                      const found = attrValues.find(at => at.id === col.id);
                                      const val = found ? found.value : '';
                                      return (
                                        <td key={i}>
                                          <Input data={col} onChange={this.onAttrChange} value={val} />
                                        </td>
                                      );
                                    })
                                  }
                                </tr>
                              );
                            })
                          }
                        </tbody>
                      </table>
                    </td>
                  </tr>
                );
              } else if (type.machineName === 'boolean') {
                const found = attrValues.find(at => at.id === attr.id);
                const val = found ? found.value : false;
                return (
                  <tr key={index}>
                    <td>{attr.name}</td>
                    <td>
                      <Checkbox
                        data={attr}
                        onChange={this.onCheckboxToggle}
                        checked={val}
                      />
                    </td>
                  </tr>
                );
              } else if (type.machineName === 'value_with_deviations') {
                const found = attrValues.find(at => at.id === attr.id);
                const val = found ? found.value : '';
                return (
                  <tr key={index}>
                    <td>{attr.name}</td>
                    <td><DeviationInput data={attr} onChange={this.onAttrChange} value={val} /></td>
                  </tr>
                );
              }
              const found = attrValues.find(at => at.id === attr.id);
              const val = found ? found.value : '';
              return (
                <tr key={index}>
                  <td>{attr.name}</td>
                  <td><Input data={attr} onChange={this.onAttrChange} value={val} /></td>
                </tr>
              );
            })
          }
        </tbody>
      </table>
    );
  }
}

const mapStateToProps = ({ document, types }) => ({ document, types });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(AttributesForm);
