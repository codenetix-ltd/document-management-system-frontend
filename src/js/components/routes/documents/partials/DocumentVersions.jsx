import moment from 'moment';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import autobind from 'autobind-decorator';
import Modal from 'react-responsive-modal';

import { DataLink } from 'Components/common/dataControls';

import {
  $$versionsFetch,
  $$versionFetch
} from 'Store/thunks/versions';

import ViewContent from './ViewContent';

@autobind
export class DocumentVersions extends Component {
  static propTypes = {
    match: PropTypes.any.isRequired,
    versions: PropTypes.any.isRequired,
    version: PropTypes.any.isRequired,
    dispatch: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = { open: false };
  }

  componentDidMount() {
    const { dispatch, match: { params: { documentID } } } = this.props;
    if (documentID) {
      $$versionsFetch(dispatch, documentID);
    }
  }

  onViewClick(versionID) {
    const { dispatch, match: { params: { documentID } } } = this.props;
    $$versionFetch(dispatch, documentID, versionID, () => {
      this.setState({ open: true });
    });
  }

  onModalClose() {
    this.setState({ open: false });
  }

  sortDesc(a, b) {
    const left = Number(a.id);
    const right = Number(b.id);
    return right - left;
  }

  render() {
    const { versions: { list }, version } = this.props;
    const rows = list.sort(this.sortDesc).map(doc => {
      return (
        <tr key={doc.id}>
          <td>{doc.id}</td>
          <td>{doc.comment}</td>
          <td>{moment.unix(doc['createdAt']).format('YYYY-MM-DD HH:MM:SS')}</td>
          <td>
            <DataLink cls="btn btn btn-success btn-xs" data={doc.id} onClick={this.onViewClick}>
              <i className="fa fa-eye" /> View version
            </DataLink>
            &nbsp;
            <button className="btn btn btn-danger btn-xs">
              <i className="fa fa-trash" />
            </button>
          </td>
        </tr>
      );
    });
    return (
      <div>
        <table className="table table-striped">
          <thead>
            <tr><td>Version name</td><td>Comment</td><td>Created</td><td>Actions</td></tr>
          </thead>
          <tbody>{rows}</tbody>
        </table>
        <Modal open={this.state.open} onClose={this.onModalClose}>
          <h4>Document version overview</h4><hr />
          <ViewContent document={{ ...version, actualVersion: version }} />
          <button type="button" className="btn btn-default" onClick={this.onModalClose}>Close</button>
        </Modal>
      </div>
    );
  }
}

const mapStateToProps = ({ versions, version }) => ({ versions, version });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(DocumentVersions);
