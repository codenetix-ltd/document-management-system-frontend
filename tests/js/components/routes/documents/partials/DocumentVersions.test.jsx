import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { DocumentVersions } from 'Routes/documents/partials/DocumentVersions';

describe('Documents Versions', () => {
  let wrapper;
  let versions;
  let match;

  beforeEach(() => {
    versions = {
      list: []
    };
    match = {
      params: {
        documentID: 1
      }
    };
    wrapper = shallow(<DocumentVersions
      dispatch={() => {}}
      match={match}
      version={{}}
      versions={versions}
    />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
