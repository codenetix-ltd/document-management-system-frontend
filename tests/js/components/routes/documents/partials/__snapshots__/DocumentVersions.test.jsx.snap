// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents Versions should render correctly 1`] = `
<div>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <td>
          Version name
        </td>
        <td>
          Comment
        </td>
        <td>
          Created
        </td>
        <td>
          Actions
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
  <Modal
    animationDuration={500}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <h4>
      Document version overview
    </h4>
    <hr />
    <DocumentViewContent
      document={
        Object {
          "actualVersion": Object {},
        }
      }
    />
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </Modal>
</div>
`;
